### PR TITLE
Adds Cuban Pete's hat to the uplink.

### PR DESCRIPTION
**Pull request by Vivalas**
Adds a special version of Cuban Pete's hat to the uplink that costs 20 TC. The hat has a one-use ability that causes the station to succumb to the Rhumba beat. Anyone who sees Pete during the 55 second duration of the ability has a 50% chance of being stunned for 10 seconds, every second (doesn't stack). They might be able to free themselves if they are lucky when the stun wears off before it starts again. If the hat is removed or Pete falls unconscious, the ability will end prematurely.

Does not affect synthetics, so the station has a little ability to fight back. 

:cl:
add: Traitors can now be the King of the Rhumba Beat himself. A new uplink item has been added.
/:cl:

### CODE REVIEW

**Comment by TheDZD** [comment]
The sound file is 816 KB, that *miiiight* be a bit much. Did you do any compressing?

**Comment by Vivalas** [comment]
Not really, I can if you want. I agree close to 1MB is a bit much. I clipped it at 55 seconds because it was a good place to cut off and seemed like the right duration.

**Comment by Fox-McCloud** [comment]
:-1: to the sound.

**Comment by Vivalas** [comment]
The sound is literally the entire PR :P

**Comment by FreeStylaLT** [comment]
Can you give us a link to the sounnnd

**Comment by Fox-McCloud** [comment]
Aside from that, a 55 second duration AoE stun is pretty ridiculous.

**Comment by FreeStylaLT** [comment]
also I have a midi of the song that's like 51 KB

**Comment by TheDZD** [comment]
@FreeStylaLT You can hit the 'View' button under the sound file in 'Files Changed', then click 'View Raw'.

**Comment by Vivalas** [comment]
Cut the sound file in half. Regarding OP-ness, a 55 second AoE stun really isn't that bad when you completely lose any aspect of stealth. This is really a meme item, no serious traitor will use this if they want greentext. I can reduce the duration to 5 seconds to make it easier to resist out of.

**Comment by Vivalas** [comment]
@FreeStylaLT  I would like to see that file.

**Comment by FreeStylaLT** [comment]
@Vivalas 
http://midishrine.com/midi/55669.mid

**Comment by Lady-Luck** [comment]
That midi sounds really awful tbh

**Comment by Vivalas** [comment]
I might consider the MIDI if the maintainers still think 450 kB is too much. I still kinda like the current one better.

**Comment by LightningLynx** [comment]
Easy baby hijacks here we come.

55 seconds is going to be an eternity if two traitors decide to collude on a hijack and team up with a hat + chainsaw or something of the sort.

This just isnt a good idea at all

**Comment by Magikarpet** [comment]
This item is crazy if you think about it like this: This item effectively allows a versatile traitor to break into a secure department, activate the AOE stun when they whip out their guns(Assume you're fighting security), and would likley have at least 1-2 guns at their disposals stolen for killing the officers. Assuming someone isn't wasteful, they can clear out security solo if they play their cards right and intelligently, and they can loot the armory. Outside of the detective, there is no one on the station who can stop an antag with access to the armory outside of security. If the antagonist murders security with the 55 second AOE stun, then it's basically good game for the station.

Compared to a holoparasite, garrotte, mini energy crossbow, or even sleeping carp? This item outclasses it in every way. 55 seconds of invincibility from just outright aoe stunning people would be cancer.

Killing the AI as a traitor even without science is relatively easy too, as all the tools outside of a welder are always in stock in the toolvend at arrivals in >90 pop rounds. With the AI dead, they almost have no opposition and they can clear out the only opposition they WOULD receive(Security) using the primary stun, and then using the armory to run around with an escape alone/murderboner's dream weapons. The TC cost may seem steep, but effective usage of your 55 seconds of AOE stun would effectively allow you to wipe out a 50+ pop server. The only opposition you might receive are mechs and the sec pod pilot, with the sec pod pilot being the biggest threat due to no pod stuns. Making it so the hat drops on stun or has a larger delay between stunning people would significantly lower the "Exterminate one department then snowball off of it" shit an antag could do with it.

**Comment by Fox-McCloud** [comment]
It's a 55 second duration permastun, from what I can see, because it's a a while loop. Yes, it doesn't impact people who aren't stunned already, but as soon as it wears off, it reprocs, so its' basically an AoE flashbang.

@Magikarpet pretty much outlined my exact issues with this.

**Comment by Vivalas** [comment]
You two bring up good points. If it's too OP to make a traitor item, perhaps I can make it admin only.

Either that or reduce the stun chance and duration.

**Comment by Lady-Luck** [comment]
I think it needs to be rethought. Maybe remove the crowd stun completely and instead make Cuban Pete lots of immunities for the duration and fill both his hands with maracas so he can't perform any other actions while the song is going on? And also a major slow debuff to himself as well.

It wouldn't be INCREDIBLY useful like that, but it would at least be fun.

**Comment by Vivalas** [comment]
I was actually considering making him not be able to attack people, so filling his hands with maracas seems fun. A slow debuff... eh... not really.

**Comment by Vivalas** [comment]
Also I'm not sure if changing it to a 25% chance of a 4 second stun is too much of a nerf.

**Comment by FreeStylaLT** [comment]
that's just a much shittier version of Mind Batterer

**Comment by Lady-Luck** [comment]
You're having trouble making this effective while making it not overpowered - so why not just ditch trying to make it effective altogether and focus on making it fun.

I could see this being a great way for a traitor to showboat.

**Comment by Vivalas** [comment]
The whole point was to have fun with it, I'm going to make it 50% chance of 10 second stun again but with the addition of maracas. That way the traitor can't do much by themselves. And can't team up with this because it would affect the other traitor as well.

**Comment by Lady-Luck** [comment]
Traitor could easily be hulk.

**Comment by Vivalas** [comment]
If traitors do team up with this, good for them. It would be pretty hilarious.

**Comment by Vivalas** [comment]
Oh shit forgot to exclude this from new cops, for obvious reasons.

**Comment by Vivalas** [comment]
Okay rebalances this thing by making it so that the wearer of the hat cannot do anything while using it (because their hands fill with maracas), and changes it back to the original stun duration. This makes it more of a meme thing than a utility thing, which is on purpose, but it still has its uses. 

Also removes it from nuke ops, because fuck that we don't need 5 cuban petes getting da fukken disk.

**Comment by Vivalas** [comment]
@Magikarpet does this address your concerns?

**Comment by FreeStylaLT** [comment]
where's the no-drop poncho forced onto exosuit slot for the duration

**Comment by FreeStylaLT** [comment]
If serious, I think this is a wonderfully fun addition, which won't get much use, but really doesn't have to be balanced to utmost scrutiny, since by its nature it's meant to be used for fun

Also, do you think the maraces would look better when facing up?

**Comment by Vivalas** [comment]
I think I'm where I want to be regarding balance. I don't care so much about balance as much as I care about gegtting this merged.

The sprites would look better facing up. Spriting isn't my forte.

File: code/modules/clothing/head/misc_special.dm [hunk 1]
```diff
@@ -232,3 +232,79 @@
...
+	var/obj/item/toy/cuban_maraca/L = new(pete)
+
+	pete.put_in_l_hand(L)
+	pete.put_in_r_hand(R)
+
+
+
+	rhumba_proc()
+
+/obj/item/clothing/head/collectable/petehat/special/proc/rhumba_proc()
```
Comment (Fox-McCloud) [inline code/modules/clothing/head/misc_special.dm]: some of this seems more appropriate for a process loop than a whitle loop, but eh. Your thoughts @TheDZD

**Comment by taukausanake** [comment]
> Also removes it from nuke ops, because fuck that we don't need 5 cuban petes getting da fukken disk.

We really do need them. You have no idea how badly we need them.
I'll wait for adminbus one day. **ONE DAY**

**Comment by Fox-McCloud** [comment]
After thinking and talking it over a bit, this seems like a bit much and a bit too gimmicky. Additional, the file size of this is rather huge; all for something that will never really be used.

**Comment by Vivalas** [comment]
#BringBackFun

Yeah I guess. :(

**Pull request closed without merging**